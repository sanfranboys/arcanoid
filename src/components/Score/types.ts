export type ScoreProps = {
  small?: boolean
}
