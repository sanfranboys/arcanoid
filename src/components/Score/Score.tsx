import React, { FC } from 'react';
import { ScoreProps } from './types';
import classNames from '../../utils';

import './Score.scss'

const Score: FC<ScoreProps> = (props) => {
  const { children, small } = props;
  const className = classNames('score', { 'score_small': small })

  return <span className={className}>{ children }</span>
}

export default Score;
