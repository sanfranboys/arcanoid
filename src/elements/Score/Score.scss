.score {
  font-size: var(--size-text-5xl);
  font-weight: bold;
  font-family: 'Agency FB', sans-serif;
  color: var(--color-gold);

  &_small {
    font-size: var(--size-text-4xl);
    color: var(--color-light-purple);
  }
}
