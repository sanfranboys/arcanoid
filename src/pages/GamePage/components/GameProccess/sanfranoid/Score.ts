import { Feature } from './Feature'

export class Score extends Feature {
  count = 0

  x = 8

  y = 20

  draw() {
    const { ctx, count, color, x, y } = this

    if (!ctx) {
      return
    }

    ctx.font = '16px Arial'
    ctx.fillStyle = color
    ctx.fillText(`Score: ${count}`, x, y)
  }

  increase() {
    this.count += 1
  }
}
