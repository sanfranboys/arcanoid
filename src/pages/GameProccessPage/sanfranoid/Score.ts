import { Feature } from './Feature'
import { Color } from './settings'

export class Score extends Feature {
  private _count = 0

  protected _x = 70

  protected _y = 20

  protected _color = Color.White

  public draw() {
    const { _ctx, _count, _color, _x, _y } = this

    _ctx.font = '16px Arial'
    _ctx.fillStyle = _color
    _ctx.fillText(`Score: ${_count}`, _x, _y)
  }

  public getScore() {
    return this._count
  }

  public increase() {
    this._count += 1
  }
}
