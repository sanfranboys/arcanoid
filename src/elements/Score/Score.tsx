import React, { FC } from 'react'
import classNames from 'utils/'
import { ScoreProps } from './types'

import './Score.scss'

const Score: FC<ScoreProps> = ({ children, small }) => {
  const className = classNames('score', { score_small: small })

  return <span className={className}>{children}</span>
}

export default Score
