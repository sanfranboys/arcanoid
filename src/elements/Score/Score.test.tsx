import React from 'react'
import { create } from 'react-test-renderer'
import Score from './Score'

describe('Score testing', () => {
  const props = {
    small: true,
    children: 'Score here',
  }

  describe('Score snapshot testing', () => {
    it('snapshot matches', () => {
      const tree = create(<Score {...props} />).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
