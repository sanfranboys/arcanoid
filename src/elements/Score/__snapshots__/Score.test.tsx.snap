// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Score testing Score snapshot testing snapshot matches 1`] = `
<span
  className="score score_small"
>
  Score here
</span>
`;
